File: server/Models/Keep.cs
using System.ComponentModel.DataAnnotations;

namespace keepr.Models;

public class Keep : RepoItem<int>
{
  [MaxLength(255)] public string Name { get; set; }
  [MaxLength(1000)] public string Description { get; set; }
  [Url, MaxLength(1000)] public string Img { get; set; }
  public int Views { get; set; }
  [MaxLength(255)] public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public Kept Kept { get; set; }
}

File: client/src/models/Keep.js
import { APIItem } from "./APIItem.js";
import { Kept } from "./Kept.js";
import { Profile } from "./Profile.js";

export class Keep extends APIItem{
  constructor(data){
super(data)
this.name = data.name
this.description = data.description
this.img = data.img
this.views = data.views
this.creatorId = data.creatorId
this.creator = new Profile(data.creator)
this.kept = new Kept(data.kept)
  }
}